Use canonical get_size_with_aspect_ratio (with max_size) from transformers.image_transforms to fix #37939

sonianuj287 opened the pull request:
Removed max_size parameter from ConditionalDetrImageProcessor
to simplify image resizing logic and fix #37939.

Used canonical get_size_with_aspect_ratio (with max_size) from transformers.image_transforms 

Tested with pytest suite and all tests are passed.

Happy to be guided further for improvement.
Thanks

sonianuj287 committed:
Use canonical get_size_with_aspect_ratio (with max_size) from transformers.image_transforms to fix #37939

Rocketknight1 commented:
cc @molbap @yonigozlan

sonianuj287 committed:
Fix import sorting/style

sonianuj287 committed:
Fix import order

molbap reviewed (approved):
Thanks for the contribution! I'm all for simplifying the image processing logic wherever we can, however this will break Backwards Compatibility on a few models (from what I checked, not many). cc @yonigozlan wdyt?

@sonianuj287 , Can you please edit the PR's name to add "🚨🚨🚨" in front so we know it's breaking? Thank you!

sonianuj287 commented:
Hi @molbap , Thanks for replying. Okay will change the PR's name. 
For the issue, the other way to resolve is to fix all function calls for this function - "get_size_with_aspect_ratio" by removing max_size parameter.

yonigozlan reviewed (commented):
Hey @sonianuj287 ! Thanks for working on this. However I'm not sure this fixes https://github.com/huggingface/transformers/issues/37939 as `max_size` is still accepted as an arg?
To deprecate it fully we need to modify the `resize` function of conditional detr. However I agree with removing `get_size_with_aspect_ratio` from the image processor and importing it from `image_transforms` as they seem to be equivalent.

We can first start with merging this PR to deprecate max_size and import `get_size_with_aspect_ratio` from `image_transforms`, and I think we can then extend this to other models which have an equivalent `get_size_with_aspect_ratio` to the  `image_transforms` one plainly written in their image processing file, along with support for `max_size` :)

yonigozlan commented:
However as @qubvel noted in #37939, we still have a lot of popular models using `max_size` on the hub, so we must be careful how we approach this. We can first remove all `get_size_with_aspect_ratio` in imaging files if they are equivalent to the `image_transforms` one, as this shouldn't cause any issues with BC

sonianuj287 commented:
HI @yonigozlan, Thanks for confirming to merge this PR. 

Let me get the task to change in other models which have an equivalent get_size_with_aspect_ratio to the image_transforms one plainly written in their image processing file, along with support for max_size.

Delighted to help you. :)

sonianuj287 commented:
Hi @yonigozlan , I resolved the conflict, and the workflows also passed. Let me know for any other changes, and we good to close. :)

yonigozlan reviewed (commented):
Hey @sonianuj287, be careful not to remove get_size_with_aspect_ratio if it's not the same as in image_transforms, for the yolos image processors, they are not the same! Also it looks like you haven't

sonianuj287 commented on src/transformers/image_processing_utils_fast.py:
```diff
@@ -23,7 +23,6 @@
 from .image_transforms import (
     convert_to_rgb,
     get_resize_output_image_size,
-    get_size_with_aspect_ratio,
```
Hey, yeah my automation pushed it along with commit, the work is still in progress. Will keep you posted. 
Thanks :)

sonianuj287 committed:
Refactor: use canonical get_size_with_aspect_ratio across image processors (except YOLOS)

This commit updates image processing utilities in multiple model processors to use the shared
transformers.image_transforms.get_size_with_aspect_ratio for consistent resizing logic and
aspect ratio handling.

YOLOS processors are intentionally left unchanged in this commit to preserve their current
behavior and avoid breaking model-specific padding/resizing assumptions. YOLOS will be updated
in a dedicated follow-up PR once compatibility is fully verified.

sonianuj287 committed:
ruff fixes

sonianuj287 committed:
Fix check_copies.py references for get_size_with_aspect_ratio to use canonical transformers.image_transforms version

sonianuj287 committed:
Merge branch 'main' into fix-conditionaldetr-remove-maxsize

yonigozlan committed:
Merge branch 'main' into fix-conditionaldetr-remove-maxsize

yonigozlan reviewed (approved):
Thanks for working on this @sonianuj287 ! Looks good to me, let's wait for the CI to pass then I'll merge :)

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_41284). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.
